modules/linear_boltzmann_solvers/executors/pi_keigen_smm.cc

```diff
@@ -0,0 +1,1122 @@
+  {
+    const auto& psi_uk_man = groupset.psi_uk_man_;
+    const auto& quadrature = groupset.quadrature_;
+    const auto num_gs_angles = quadrature->omegas_.size();
+    const auto wt_sum =
+      std::accumulate(quadrature->weights_.begin(), quadrature->weights_.end(), 0.0);
+
+    const auto first_grp = groupset.groups_.front().id_;
+    const auto num_gs_groups = groupset.groups_.size();
+
+    // Loop over cells
+    for (const auto& cell : grid.local_cells)
+    {
+      const auto& transport_view = transport_views[cell.local_id_];
+      const auto& cell_mapping = pwld.GetCellMapping(cell);
+
+      // Compute node-wise, groupset wise tensors
+      for (int i = 0; i < transport_view.NumNodes(); ++i)
+      {
+        for (int gsg = 0; gsg < num_gs_groups; ++gsg)
+        {
+          const auto g = first_grp + gsg;
+
+          // Get the tensor, set it to zero
+          const auto dof_map = pwld.MapDOFLocal(cell, i, tensor_uk_man_, g, 0);
+          double* T = &local_tensors[dof_map];
+
+          // Perform the quadrature integration
+          for (int n = 0; n < num_gs_angles; ++n)
+          {
+            const auto& omega = quadrature->omegas_[n];
+            const auto& wt = quadrature->weights_[n];
+
+            const auto psi_dof = pwld.MapDOFLocal(cell, i, psi_uk_man, n, gsg);
+            const auto coeff = wt * psi[gs][psi_dof];
+
+            for (int k = 0; k < dimension_; ++k)
+            {
+              for (int l = 0; l < dimension_; ++l)
+                T[k * dimension_ + l] +=
+                  coeff * omega[dimension_ > 1 ? k : 2] * omega[dimension_ > 1 ? l : 2];
+              T[k * dimension_ + k] -= coeff / 3.0;
+            }
+          } // for angle n
+        }   // for groupset group gsg
+      }     // for node i
+
+      // Loop over cell faces
+      int f = 0;
+      for (const auto& face : cell.faces_)
+      {
+        if (not face.has_neighbor_)
+        {
+          const auto num_face_nodes = cell_mapping.NumFaceNodes(f);
+          for (int fi = 0; fi < num_face_nodes; ++fi)
+          {
+            const auto i = cell_mapping.MapFaceNode(f, fi);
+            const auto imap = pwld.MapDOFLocal(cell, i);
+
+            // Compute the diffusion boundary factor. This is simply the
+            // computation of the quadrature of \Omega \cdot \hat{n} divided
+            // by the sum of the quadrature weights. In most cases, this should
+            // be equivalent to half.
+            if (bndry_factors_.count(imap) == 0)
+            {
+              bndry_factors_[imap].resize(num_groupsets, 0.0);
+
+              double val = 0.0;
+              for (int n = 0; n < num_gs_angles; ++n)
+                val +=
+                  quadrature->weights_[n] * std::fabs(quadrature->omegas_[n].Dot(face.normal_));
+              bndry_factors_[imap][gs] = val / wt_sum;
+            }
+            //            const auto& bfac = bndry_factors_[imap][gs];
```

Comment:
is this supposed to be commetned out and left here?

Reply:
For right now, yes. There is a bit of a story behind this that I will explain in the next comment.